packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1392,7 +1421,13 @@ class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetec
     // because it's 2 single taps, each of which may do different things depending
     // on whether it's a single tap, the first tap of a double tap, the second
     // tap held down, a clean double tap etc.
-    if (_doubleTapTimer != null && _isWithinDoubleTapTolerance(details.globalPosition)) {
+    if (_tripleTapTimer != null && _isWithinDoubleTapTolerance(details.globalPosition)) {
+      widget.onTripleTapDown?.call(details);
+
+      _tripleTapTimer!.cancel();
```

Comment:
I'm not sure if this edge case would ever be triggered, but wondering if we want to cancel/reset all these timers gestures like force press, long press, drag, etc are fired . Once you've started any of these, I think there's no longer the possibility that a double or triple tap would be fired.

Related to that, it feels like there are a lot of states to keep track of. I am not super familiar with the GestureRecognizer system, but is it possible to lean on that to deal with all the edge cases?

Reply:
nit: can you generalize this a bit (use a counter instead of 2 `bool` variables and reuse the timer if possible)? 

Also I tested on Android and iOS. It seems on iOS if you keep tapping the paragraph selection persists whereas on Android the selection keeps changing. I'm not sure people develop muscle memory off of that, but I like the iOS behavior a bit more since if you tap 4 times in quick succession it's usually by mistake. WDYT?

Reply:
Yes sure. I was originally restructuring too much in this PR but a single timer and counter sounds easier to manage.

I prefer the iOS behavior as well. And the same happens when I'm typing into the TextArea on GItHub.

Reply:
> I think you should be able to detect such edge cases in `onTapCancel`?

Wasn't familiar with the behavior of that, will look into it!

Reply:
@LongCatIsLooong I noticed that in the build method we are returning a `RawGestureDetector` with factories. Would it be feasible to introduce a `StackedMultiTapGestureDetector` (or something like) that so it plugs into the existing gesture detection structure?

Or is it not done this way because we need to be able to fire multiple events?

I'll stick to the current goal though, so I don't increase the scope :)

Reply:
I generalized it a bit more so there's fewer cases. The numbers 2 & 3 are still hardcoded, but I figured we can generalize even more if we ever need to add quadruple click or more.